### DISCUSSION TITLE
What's the recommended way to type OneToOne fields?

### DISCUSSION THREAD

**Discussion by rvanlaar** [sbdchd/django-types#267] [Q&A]
We have a user model and different types of users. Very simplified, we have users, and some users can have API access. Then there's a OneToOne from an `APIUser` model to the `User` model.

```python
class User(AbstractUser):
    pass
    
class APIUser:
    user = models.OneToOneField(
        settings.AUTH_USER_MODEL, on_delete=models.CASCADE, primary_key=True
    )
    api_key = models.CharField()
```

When a `User` is an `APIUser` I can use `user.apiuser`. Which has a type of `APIUser`.
However when it's the User is not an APIUser then Django will raise `RelatedObjectDoesNotExist` when trying to access `apiuser`. 

What's the recommended way to type hint this on the User model?

**Comment by sbdchd** [comment]
does `APIUser` inherit from `User`? 

trying to figure out how they connect -- usually a generic somewhere can help

**Comment by rvanlaar** [comment]
As shown in the short example code, APIUser doesn't inherit.
